docs/authentication/fb-google-oauth2.html

```diff
@@ -1,28 +1,360 @@
+       For a mobile application, the user must authenticate themselves on the device, send the access token to the server and have the token authenticated using a Credentials token plugin.</p>
+  </div>
+  <h2 class="heading-2">Prerequisites</h2>
+  <ul class="plain-list">
+    <li><a onclick="setActiveSidebarElementForParent('create-server')" href="../getting-started/create-server.html#">Kitura Server:</a> Learn how to create one in our Getting Started guide.</li>
+    <li><a onclick="setActiveSidebarElementForParent('style-guide')" href="../getting-started/style-guide.html#">Kitura Style Guide:</a> Follow our guide to see how a production ready server should be structured.</li>
+  </ul>
+  <div class="underline"></div>
+  <h2 class="heading-2"><span class="blue-text">Step 1:</span> Create your third party OAuth2 application</h2>
+  <p class="block-text">For this guide we need a third party OAuth2 provider who will authenticate the user.
+       Select either Facebook or Google below and we will show you how to create the OAuth2 provider for that platform.</p>
+   <section class="tab">
+     <div class="tab-buttons">
+         <button id="facebook-button" type="button" name="button" class="tablinks facebookTab" onclick="openPlugin('facebookTab', '')">Facebook</button>
+         <button type="button" name="button" class="tablinks googleTab" onclick="openPlugin('googleTab', '')">Google</button>
+     </div>
+     <div class="info">
+         <p>The code examples for the rest of the guide will be updated to match the selected OAuth2 provider.</p>
+     </div>
+     </section>
+     <div class="tabcontent facebookTab">
+         <ol>
+         <li><p>Log in/sign up to <a target="_blank" href="https://developers.facebook.com/apps/">Facebook developers.</a><p></li>
+         <li><p>Create a new Facebook App with a name of your choice.</p></li>
+         <li><p>From the dashboard, add the Facebook login product.</p></li>
+         <li><p>Inside Facebook login settings, add `http://localhost:8080/oauth2/facebook` to the Valid OAuth Redirect URIs.</p></li>
+         <li><p>Inside the App settings, note down the App ID and App Secret. We will need them later.</p></li>
+         </ol>
+     </div>
+     <div class="tabcontent googleTab">
+         <ol>
+         <li><p>Follow the Google <a target="_blank" href="https://support.google.com/cloud/answer/6158849?hl=en">Setting up OAuth 2.0</a> guide.</p></li>
+         <li><p>Once you have created your client ID, add `http://localhost:8080/oauth2/google` to your authorized redirect URIs.</p></li>
+         <li><p>Note down the client ID and client secret. We will need them later.</p></li>
+         </ol>
+     </div>
+  <div class="underline"></div>
+  <h2 class="heading-2"><span class="blue-text">Step 2:</span> Create the OAuth2 routes</h2>
+  <p>Firstly, open your `Application.swift` file in your default text editor (or Xcode if you prefer):</p>
```

Comment:
It would be helpful to have a description of what we're going to do in this step and why, as when you're opening the Application.swift file and putting the function into postInit() you don't know why you're doing that.